File: tests/tester.py
#!/usr/bin/env python

import sys, os.path

try:
    from urllib.parse import parse_qsl
except ImportError:
    from urlparse import parse_qsl

PY3 = (sys.version_info >= (3, 0, 0))

sys.path.insert(0, os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__))), 'python'))
from om import om

consumer = (sys.argv[1], sys.argv[2])
url = sys.argv[3]

if PY3:
    params = [(k, v) for k, v in parse_qsl(sys.argv[4])]
else:
    params = [(k.decode('utf-8'), v.decode('utf-8')) for k, v in parse_qsl(sys.argv[4])]

if sys.argv[5] != '' and sys.argv[6] != '':
  token = (sys.argv[5], sys.argv[6])
else:
  token = None
method = sys.argv[7]
if sys.argv[8] != '':
  realm = sys.argv[8]
else:
  realm = None
timestamp = sys.argv[9]
nonce = sys.argv[10]
sys.stdout.write(om(consumer, url, params, token, method, realm, timestamp, nonce))

File: tests/Tester.java
import com.rdio.simple.Om;
import com.rdio.simple.Parameters;

public final class Tester {
  public static void main(String[] args) {
    String token = null;
    String tokenSecret = null;
    if (args[4].length() > 0 && args[5].length() > 0) {
      token = args[4];
      tokenSecret = args[5];
    }
    String realm = null;
    if (args[7].length() > 0) {
      realm = args[7];
    }
    System.out.print(Om.sign(
        args[0], args[1], // consumer
        args[2], // url
        Parameters.fromPercentEncoded(args[3]),
        token, tokenSecret,
        args[6], // method
        realm,
        args[8], args[9] // timestamp & nonce
    ));
  }
}